tgui/packages/tgui/interfaces/Hypospray.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author CodeJester (https://github.com/codeJester27)
+ * @license ISC
+ */
+
+import { useBackend } from "../backend";
+import { Button, Section, Slider } from "../components";
+import { Window } from '../layouts';
+import { ReagentGraph, ReagentList } from './common/ReagentInfo';
+import { glitch } from './common/stringUtils';
+
+export const Hypospray = (_props, context) => {
+  const { act, data } = useBackend(context);
+  const { emagged, injectionAmount, reagentData } = data;
+
+  return (
+    <Window
+      width={320}
+      height={300}
+      theme={emagged ? "syndicate" : "nanotrasen"}>
+      <Window.Content>
+        <Section title={emagged ? glitch("Contents", 3) : "Contents"}
+          buttons={
+            <Button
+              icon="times"
+              color="red"
+              disabled={!reagentData.totalVolume}
+              onClick={() => act('dump')}
+            >
+              Dump
+            </Button>
+          }>
+          <ReagentGraph container={reagentData} />
+          <ReagentList container={reagentData} />
+        </Section>
+        <Section title="Injection Amount">
+          <Slider
+            value={injectionAmount}
+            format={value => value + "u"}
+            minValue={1}
+            maxValue={reagentData.maxVolume}
+            step={1}
+            stepPixelSize={10}
```

Comment:
I haven't personally used the `Slider` component much, I understand how this works with a `Knob` component, is this much the same? I'm unclear on how this would interact with a Slider in that I would assume dragging the slider from left to right would move the amount from minimum to maximum.

Reply:
yes

Reply:
Right, so unsure how `stepPixelSize` interacts if you're just dragging it from left to right?

Reply:
Really just acts like a horizontal `Knob`, but it looks more like a progress bar. Both of them are `DraggableControl` components.

Reply:
I'll pull the branch down locally and mess with.

Reply:
In case you haven't already, there is a section of the Kitchen Sink where you can mess with a slider